### PR TITLE
fix(meshmetric): add missing timestamp in mapper

### PR DESCRIPTION
**Pull request by slonka**
Otherwise we'll get:

```
Permanent error: error exporting items, request to https://otlp-gateway-prod-eu-west-2.grafana.net/otlp/v1/metrics responded with HTTP Status Code 400, Message=send data to ingesters: failed pushing to ingester ingester-zone-a-103: user=REDACTED: the sample has been rejected because another sample with the same timestamp, but a different value, has already been ingested (err-mimir-sample-duplicate-timestamp). The affected sample has timestamp 1970-01-01T00:00:00Z and is from series envoy_cluster_upstream_rq_time_bucket...
```

while trying to export to grafana cloud.

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues -- not reported yet
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kuma-dp/pkg/dataplane/metrics/testdata/otel/counter.golden.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
             }
           ],
           "StartTime": "0001-01-01T00:00:00Z",
-          "Time": "0001-01-01T00:00:00Z",
+          "Time": "2014-03-17T15:26:03+01:00",
```
Comment (slonka) [inline app/kuma-dp/pkg/dataplane/metrics/testdata/otel/counter.golden.json:47]: this is fallback time

File: app/kuma-dp/pkg/dataplane/metrics/testdata/otel/counter.golden.json [hunk 2]
```diff
@@ -86,7 +86,7 @@
             }
           ],
           "StartTime": "0001-01-01T00:00:00Z",
-          "Time": "0001-01-01T00:00:00Z",
+          "Time": "2024-01-01T01:01:01.000000001Z",
```
Comment (slonka) [inline app/kuma-dp/pkg/dataplane/metrics/testdata/otel/counter.golden.json:89]: this is fallback / request time from https://github.com/kumahq/kuma/pull/10966/files#r1689914172

File: app/kuma-dp/pkg/dataplane/metrics/metrics_producer.go [hunk 3]
```diff
@@ -100,6 +101,7 @@ func (ap *AggregatedProducer) fetchStats(ctx context.Context, app ApplicationToS
 		return nil
 	}
 	defer resp.Body.Close()
+	requestTime := time.Now()
```
[the remark this answers is not in the document]
Comment (Automaat) [reply app/kuma-dp/pkg/dataplane/metrics/metrics_producer.go:104]: I think that one time for whole request is ok, also this is fallback so this should not be used that often

**Review by Automaat — APPROVED** [review]

**Comment by slonka** [comment]
@lahabana should this be backported?

File: app/kuma-dp/pkg/dataplane/metrics/metrics_format_mapper.go [hunk 4]
```diff
@@ -49,24 +50,26 @@ func FromPrometheusMetrics(appMetrics map[string]*io_prometheus_client.MetricFam
...
 }
 
-func summaryDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue) []metricdata.SummaryDataPoint {
+func summaryDataPoints(prometheusData []*io_prometheus_client.Metric, extraAttributes []attribute.KeyValue, requestTime time.Time) []metricdata.SummaryDataPoint {
 	var dataPoints []metricdata.SummaryDataPoint
 	for _, metric := range prometheusData {
 		attributes := createOpenTelemetryAttributes(metric.Label, extraAttributes)
 		dataPoints = append(dataPoints, metricdata.SummaryDataPoint{
 			Attributes:     attributes,
+			Time:           getTimeOrFallback(metric.TimestampMs, requestTime),
```
Comment (lukidzi) [inline app/kuma-dp/pkg/dataplane/metrics/metrics_format_mapper.go:72]: Question, don't we need to set `StartTime` aswell?

Comment (slonka) [reply app/kuma-dp/pkg/dataplane/metrics/metrics_format_mapper.go:72]: https://opentelemetry.io/docs/specs/otel/metrics/data-model/#gauge as otel docs mention this is optional 
image

**Pull request merged**